core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -0,0 +1,137 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.model.Frame;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.ToString;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.rnorth.ducttape.TimeoutException;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * 'Sidekick container' with the sole purpose of recording the VNC screen output from another container.
+ *
+ */
+@Getter
+@ToString
+public class VncRecordingContainer extends GenericContainer<VncRecordingContainer> {
+
+    private static final String RECORDING_FILE_NAME = "/screen.flv";
+
+    public static final String DEFAULT_VNC_PASSWORD = "secret";
+
+    public static final int DEFAULT_VNC_PORT = 5900;
+
+    private final String targetNetworkAlias;
+
+    private String vncPassword = DEFAULT_VNC_PASSWORD;
+
+    private int vncPort = 5900;
+
+    private int frameRate = 30;
+
+    public VncRecordingContainer(@NonNull GenericContainer<?> targetContainer) throws IllegalStateException {
+        this(
+                targetContainer.getNetwork(),
+                targetContainer.getNetworkAliases().stream()
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("Target container must have a network alias"))
+        );
+    }
+
+    /**
+     * Create a sidekick container and attach it to another container. The VNC output of that container will be recorded.
+     */
+    public VncRecordingContainer(@NonNull Network network, @NonNull String targetNetworkAlias) throws IllegalStateException {
+        super(TestcontainersConfiguration.getInstance().getVncRecordedContainerImage());
+
+        this.targetNetworkAlias = targetNetworkAlias;
+        withNetwork(network);
+
+        waitingFor(new AbstractWaitStrategy() {
```

Comment:
Our standard LogWaitStrategy didn't work here and also requires a Regexp while here `contains` is more than enough. Also, in debug mode `vncrec.py` produces a lot of output, makes sense to do as little as possible here

Reply:
Perhaps we could/should ship a log wait strategy that is based on `contains` anyway. It seems to me that `LogWaitStrategy` is used for contains semantics more often than for a full blown regex.

Not saying we have to do that now, but keen for your thoughts.

Reply:
I see your point not needing a Regex, but why didn't `LogWaitStrategy` work?